packages/web-app-files/src/store/mutations.js

```diff
@@ -324,8 +324,11 @@ export default {
 // eslint-disable-next-line camelcase
 function $_upsertResource(state, resource, allowInsert) {
   const files = [...state.files]
-  const index = files.findIndex((r) => r.id === resource.id)
-  const found = index > -1
+  let index = files.findIndex((r) => r.id === resource.id)
+  if(resource.webDavPath && resource.webDavPath.length) {
```

Comment:
Please add `index === -1` as an additional condition to the `if` clause. You check that in the body and basically do nothing in that case, so you could as well skip the block entirely, which makes it easier to read.

Reply:
is it possible at all the the `webDavPath` is empty?

Reply:
yes sadly in some unittests it can happen that webDavPath is empty